src/RequestHandler/LoginHandlerTrait.php

```diff
@@ -101,7 +85,7 @@ protected function createStartLoginResponse(
     protected function verifyLoginRequest(StoreInterface $store, HTTPRequest $request): Result
     {
         $method = $store->getMethod();
-        $methodInstance = $method ? $this->getMethodRegistry()->getMethodByURLSegment($method) : null;
+        $methodInstance = $method ? MethodRegistry::singleton()->getMethodByURLSegment($method) : null;
```

Comment:
Personally I prefer it the way it was. You can always rely on `getMethodRegistry()` returning a MethodRegistry, which also makes it easier to mock in tests, whereas now we have a little more business logic in the methods for this instead. I don't feel strongly enough about this specific example to request you to change it back, but just voicing my thoughts.

Reply:
I also prefer it the way it was but the problem is that it's not clean to do traits in this way. I can't define the `getMethodRegistry` on this trait because I need it on the other trait too, and method conflicts on traits cannot be resolved. There's not really any good way to do method inheritance. I played around with it for a bit and gave up